### ISSUE TITLE
Does this still support hyperclick for 'go to definition'?

### ISSUE THREAD

**Issue by jonboiser** [atom-community/ide-python#257]
I'm wondering if this package provides support for hyperclick, since I've seen it mentioned in some older issues (https://github.com/atom-community/ide-python/issues/143) and is mentioned in the [readme for atom-ide-definitions](https://github.com/atom-community/atom-ide-definitions). 

Nor do I see an implementation for a function like `getSuggestionForWord` [as in the autocomplete-python package](https://github.com/autocomplete-python/autocomplete-python/blob/3151d01238144e17285c6f0f8843271fc3b66c52/lib/hyperclick-provider.coffee)

Currently, I _think_ I have all the packages to make this work:

* atom-ide-definitions
* atom-ide-base
* atom-ide-hyperclick
* ide-python

But I can only use the "go to definition" command from the command palette.

**Comment by aminya** [comment]
Doesn't CTRL+Click work for you?

![hyperclick](https://user-images.githubusercontent.com/16418197/103719049-ce984200-4f8d-11eb-93f6-0e9e81ebced1.gif)

**Comment by jonboiser** [comment]
I use CMD-click on mac, and that doesn't work. Also running `atom-ide-hyperlick:confirm-cursor` from the command palette doesn't work when I'm view Python code (on JS and Vue code, for which I have hyperclick providers, CMD+click still works).

Here are my installed packages if that's helpful:

```
Community Packages (41) /Users/jon/.atom/packages
├── atom-ide-base@2.1.1
├── atom-ide-definitions@0.4.0
├── atom-ide-hyperclick@1.0.9
├── atom-ide-outline@1.15.5
├── atom-ide-ui@0.13.0 (disabled)
├── atom-ide-vue@1.4.3 (disabled)
├── atom-inline-blame@0.0.10 (disabled)
├── atom-wrap-in-tag@0.6.0 (disabled)
├── autocomplete-modules@2.3.0
├── autocomplete-paths@2.12.15
├── autocomplete-python@1.16.0 (disabled)
├── busy-signal@2.0.1
├── choose-pane@0.8.0
├── cursor-history@0.14.0
├── demo-mode@0.4.1
├── dracula-night-syntax@0.5.0
├── dracula-syntax@3.0.2
├── editorconfig@2.6.1
├── hyperclick@0.1.5 (disabled)
├── ide-python@1.7.2
├── inline-git-diff@2.4.0
├── intentions@1.1.5
├── js-hyperclick@1.19.0
├── language-gherkin@1.0.4
├── language-restructuredtext@1.1.2
├── language-vue@0.26.0
├── linter@3.2.3
├── linter-eslint@8.5.5
├── linter-flake8@2.4.0
├── linter-ui-default@2.3.5
├── narrow@0.65.2
├── paner@1.0.0
├── pigments@0.40.6
├── prettier-atom@0.60.1
├── project-manager@3.3.8
├── pure-syntax@2.0.0
├── quick-highlight@0.13.0
├── tone-syntax@0.2.2
├── tree-view-autoresize@2.3.0
├── vim-mode-plus@1.36.4
└── vim-mode-plus-move-selected-text@0.12.0
```

**Comment by aminya** [comment]
@UziTech any thoughts?

**Comment by jonboiser** [comment]
> Oh, by the way, you have the lsp installed, right?

Yes, it is installed globally on my system, but not necessarily in the virtualenvs I use for my projects, if that's relevant.

**Comment by jonboiser** [comment]
I just updated my config for `atom-ide-definitions`, which seems to help
```
  "atom-ide-definitions":
    clickGrammarScopes: [
      "source.python"
      "text.html.vue"
      "source.js"
    ]
```

But it's not as performant as my experience with `autocomplete-python` (nearly instantaneous when clicking to definition). Here's a GIF that shows how slowly it goes. What isn't shown is that the MacOS spinner icon appears where the cursor is, above the blue screen of lines.

![CleanShot 2021-01-05 at 19 18 18](https://user-images.githubusercontent.com/10248067/103725282-18cbf400-4f8b-11eb-9a92-e8df0eb8ae69.gif)

**Comment by aminya** [comment]
> I just updated my config for atom-ide-definitions, which seems to help

Hmm. It should work without adding that, I need to check. 

> But it's not as performant as my experience with

We can diagnose the performance. The Python LSP is generally slow because it is written in Python. If there are Python LSPs written in a more performant language, we can add the support for that.

**Comment by jonboiser** [comment]
Is there a way to configure the definition provider? I notice that sometimes the definitions go very deep, like into the site_packages directory and into things like Django source code. Maybe it can be configured to not do such a deep search for definitions.
